Implement Pause/Resume Button Functionality

ColemanMitch opened the pull request.

dwilliams27 commented on backend/server.js:
```diff
@@ -1,48 +1,55 @@
-let express = require('express')
-let request = require('request')
-let querystring = require('querystring')
+let express = require("express");
```
New linter? Or did you manually change all dat

ColemanMitch replied:
New linter that auto formats .js files

dwilliams27 commented on src/Timer.tsx:
```diff
@@ -45,24 +36,35 @@ class Timer extends React.Component<TimerProps, TimerState> {
+    }
+    else { // if it's paused make sure ticking is false
+      this.setState({
+        ticking: false
```
Is ticking ever used logically differently than paused? Or could ticking be removed and we only refer to props.paused instead?

ColemanMitch replied:
We would want to  stop ticking all together once we actually reach the max song count. 

That hasn't been implemented yet, so lemme log that as another potential issue.

dwilliams27 commented on src/App.tsx:
```diff
@@ -128,6 +131,33 @@ class App extends Component<void, AppState> {
     });
   }
 
+  pauseCurrentPlayback(): void {
```
Could probably combine these 2 methods into 1 like:

```
adjustPlayback(paused: boolean): void {
   this.setState({paused})
    (paused ? this.spotifyService.pauseCurrentPlayback() : this.spotifyService.resumeCurrentPlayback()).then(res => {
      res.body?.getReader().read().then(body => {
        if(body?.done) {
          setTimeout(() => this.fetchCurrentlyPlaying(), 1000);
        }
      });
    });
  }
```

dwilliams27 replied:
Could even make it not have any parameters and just invert what's currently the state of `paused` 🤔

ColemanMitch replied:
You thinking something like?

```this.setState({ paused: !paused})```

dwilliams27 replied:
Yeah could go with 
```
invertPlayback(): void {
   
    (!this.state.paused ? this.spotifyService.pauseCurrentPlayback() : this.spotifyService.resumeCurrentPlayback()).then(res => {
      res.body?.getReader().read().then(body => {
        if(body?.done) {
          setTimeout(() => this.fetchCurrentlyPlaying(), 1000);
        }
      });
    });
    this.setState({pause: !this.state.paused})
  }
  ```
  
 Though doesn't really matter, up to you my man

dwilliams27 commented on src/Timer.tsx:
```diff
@@ -45,24 +36,35 @@ class Timer extends React.Component<TimerProps, TimerState> {
   }
 
   tick = async () => {
+    console.log(this.props, this.state);
```
Assuming WIP, but probably dont want to print the state every tick lol

ColemanMitch replied:
Good catch

dwilliams27 commented on src/Timer.tsx:
```diff
@@ -45,24 +36,35 @@ class Timer extends React.Component<TimerProps, TimerState> {
-        });
-        return;
+    if (!this.props.paused) { // if it's playing (not paused)
+        if(this.state.time.seconds === 0) {
```
Minor, but indentation a bit off here

ColemanMitch commented:
Major of diffs are from `package-lock.json`. Ready for re-review whenever @dwilliams27

dwilliams27 reviewed (approved):
LGTM, up to you about changing the pause/play methods. You're good to merge it whenever 👍

dwilliams27 commented:
Actually you may want to remove the changes to the backend from this PR, I cleaned it up and reworked it a bit in [this PR](https://github.com/ColemanMitch/Spowerfy-Redux/pull/25).

ColemanMitch commented:
> Actually you may want to remove the changes to the backend from this PR, I cleaned it up and reworked it a bit in [this PR](https://github.com/ColemanMitch/Spowerfy-Redux/pull/25).

~~Shweet, I can revert the changes to `server.js` since they were just linting 😎~~

Actually wait nvm. We can just keep as is since they're just linting changes
